policies/testing.md

```diff
@@ -0,0 +1,42 @@
+Testing Policy
+==============
+
+This applies to all stable and development branches of the main code repository.
+Except where noted below:
+
+- All Pull Requests adding new functionality to the applications, libraries,
+providers or engines must include suitable tests.
+- All Pull Requests fixing a functional behavior defect in the applications,
+libraries, providers or engines must include a test for that defect.
+
+Pull Requests that do not change the functional behaviour of the applications,
+libraries, providers or engines do not require tests to be added. For example
+the following types of changes do not require tests:
+ - Changes to comments, formatting, internal naming or similar
+ - Fixes for performance
+ - Refactoring
+
+To be explicit the above statement means that tests are not required for changes
+in:
+ - Documentation (including CHANGES.md/NEWS.md)
+ - The test suite
+ - perl utilities
+ - Include files
+ - Build system
+ - Demos
+
+Pull Requests that only affect minor cosmetic output do not require tests
+
+Performance Testing
+-------------------
+
+Performance testing should be performed automatically via CI on a regular basis
+for certain defined components.
+
+Components that should be performance tested via CI include:
+ - openssl speed for all supported algorithms
+ - openssl s_time
```

Comment:
This seems to mention implementation details rather than what we want to do. I prefer that the policy juts mentions what kind of things should be tested.

Reply:
I've reworded this and dropped specific mention of speed and s_time